### ISSUE TITLE
OrientDB.RequestError on update

### ISSUE THREAD

**Issue by dmarcelino** [appscot/waterline-orientdb#43]
Post
``` javascript
module.exports = {

  attributes: {
    title: {
        type: 'string'
    },
    slug: {
        type: 'string'
    },
    editorialPriority: {
        type: 'string'
    },
    sectionPriority: {
        type: 'string'
    },
    html: {
        type: 'string'
    },
    editor_html: {
        type: 'string'
    },
    featureImage: {
        model: 'image'
    },
    area: {
        type: 'string'
    },
    excerpt: {
        type: 'string'
    },
    content: {
        type: 'string',
    },
    publicationDate: {
        type: 'datetime'
    },
    categories:{
        collection: 'category',
        through: 'post_category',
        via: 'post',
        dominant: true
    },
    author:{
        model:'author'
    },
    status:{
        model:'postStatus'
    },
    address: {
        type: 'string'
    },
    addressReference: {
        type: 'string'
    },
    latitude: {
        type: 'float'
    },
    longitude: {
        type: 'float'
    }
  }
}
```

Image
``` javascript
module.exports = {

  attributes: {
    file: {
        type: 'json',
        isFile: true,
        required: true
    },
    footer: {
        type: 'string'
    },
    author: {
        model: 'author'
    },
    area: {
        type: 'string'
    },
    isCrop: {
        type: 'boolean'
    },
    parent: {
        model: 'image'
    },
    crops: {
        collection: 'image',
        via: 'parent'
    }
  }
};
```

Error:
``` javascript
{ error: 'E_UNKNOWN',
  status: 500,
  summary: 'Encountered an unexpected error',
  raw: 
   { name: 'OrientDB.RequestError',
     message: 'Error on execution of command: sql.select from image WHERE   @rid = #15:0',
     data: {},
     previous: [ [Object] ],
     id: 1,
     type: 'com.orientechnologies.orient.core.exception.OCommandExecutionException',
     hasMore: 1,
     __stackCleaned__: true } }
```

Raised by @rodrigorn  on :point_up: [February 13 2015 11:20 PM](https://gitter.im/appscot/waterline-orientdb?at=54de86b9d44034aa04b92bcb)

Related: [stackoverflow.com: Node.js waterline-orientdb update fail](http://stackoverflow.com/questions/27564558/node-js-waterline-orientdb-update-fail)

**Comment by dmarcelino** [comment]
Hi @rodrigorn,

Regarding my previous [comment on gitter](https://gitter.im/appscot/waterline-orientdb?at=54e3f0b2c8e106d730c9faf2):
> I encourage you to further checkout branch 43-OrientDB_RequestError and try to reproduce the issue
then it will be easier to help you
> the only other thing we haven't tried is exposing waterline-orientdb logs. You can do that by adding the DEBUG env variable in your console such as export DEBUG=$DEBUG,waterline-orientdb:*. Try that and see if you get any ERRORs in the console.

Have you had a chance to try any of that?

**Comment by dmarcelino** [comment]
@rodrigorn, I was able to reproduce [stackoverflow.com: Node.js waterline-orientdb update fail](http://stackoverflow.com/questions/27564558/node-js-waterline-orientdb-update-fail) issue which was similar to yours. I've confirmed that with current master's code it's breaking. The good news is that future v0.10.40 branch seems to fix it. Would you mind give it a go and see if it fixes your issue as well?
``` shell
npm install appscot/waterline-orientdb#refactor_collection
```
Just be careful to use [`migrate: safe`](https://github.com/appscot/waterline-orientdb/tree/refactor_collection#waterline-orientdb).

**Comment by dmarcelino** [comment]
Fixed by #46
